### PR TITLE
Bug 1314708: Add testing login credentials, add dashboard tests

### PR DESCRIPTION
**Pull request by stephaniehobson**
- testing usernames and passwords
- page object for logging into admin
- test "sign in" forwards to github
- tests for revision dashboard

Testing usernames will be part of sample database or can be configured manually on dev machines. They will not be on staging or production.

Use `-m "not login"` to exclude tests requiring a login.

Not all login tests are destructive.

### CODE REVIEW

**Comment by stephaniehobson** [comment]
@m8ttyB R? Also, you will need to update the Jenkins command to exclude the login tests.

File: docs/tests-ui.rst [hunk 1]
```diff
@@ -140,7 +140,8 @@ Marks
 * ``login``
 
   These tests require the testing accounts to exist on the target site. For
-  security reasons these accounts will not be on production.
+  security reasons these accounts will not be on production. Exclude these tests
+  with ```-m 'not login'``
```
Comment (m8ttyB) [inline docs/tests-ui.rst]: Let's be more specific and note that these tests will only run on a local dev environment.

File: tests/pages/dashboard.py [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+    @property
+    def dashboard_not_overflowing(self):
+        crawlBar = self.selenium.execute_script("return document.documentElement.scrollWidth>document.documentElement.clientWidth;")
+        return not crawlBar
+
+class DashboardRow(Region):
+
+    _root_locator = (By.CSS_SELECTOR, '.dashboard-row')
+    _ban_ip_locator = (By.CSS_SELECTOR, '.dashboard-ban-ip-link')
+    _spam_ham_button_locator = (By.CSS_SELECTOR, '.spam-ham-button')
```
Comment (m8ttyB) [inline tests/pages/dashboard.py]: Nice clean locators!

File: tests/pages/dashboard.py [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+        # wait for notification in tray
+        self.wait.until(lambda s: len(self.find_elements(*self._first_notification_locator)) > 0)
+        # form editable when ajax response arrives
+        self.wait.until(lambda s: 'disabled' not in revsion_filter_form.get_attribute('class'))
+        # wait for notification to close
+        self.wait.until(lambda s: 'closed' in self.find_element(*self._first_notification_locator).get_attribute('class'))
+        # revsion-page-value updates to 1
+        self.wait.until(lambda s: int(self.find_element(*self._revision_page_input).get_attribute('value')) == 1)
+        # opacity maniulation finishes
+        self.wait.until(lambda s: 'opacity' not in self.find_element(*self._parent_locator).get_attribute('style'))
```
Comment (m8ttyB) [inline tests/pages/dashboard.py]: Nice usage of `wait` statements to ensure the behavior of the page matches your expectations.

**Comment by stephaniehobson** [comment]
Minor nit, picked. ☑️

**Pull request merged**